packages/cosmic-swingset/src/helpers/interceptor.js

```diff
@@ -0,0 +1,44 @@
+// @ts-check
+import { Far } from '@endo/far';
+
+export const makeJsonRPCInterceptor = dispatch => {
+  const idToInterceptor = new Map();
+  const service = Far('jsonRPCInterceptor', {
+    resolve(res) {
+      for (const [id, interceptor] of idToInterceptor.entries()) {
+        idToInterceptor.delete(id);
+        interceptor({ id, result: String(res) });
```

Comment:
Under what circumstances would we want reaching the end of input (i.e., `makeReadlineLoop` exiting successfully) to result in fulfilling outstanding requests?
```suggestion
        interceptor({ id, error: 'input ended with no reply' });
```